### ISSUE TITLE
Steam continually writing to disk, no downloads running

### ISSUE THREAD

**Issue by MadMinstrel** [ValveSoftware/steam-for-linux#8237]
#### Your system information

* Steam client version: Oct 13 2021, at 19:47:24
* Distribution (e.g. Ubuntu): Manjaro XFCE
* Opted into Steam client beta?: [Yes/No] Both.
* Have you checked for system updates?: [Yes/No] Yes.

#### Please describe your issue in as much detail as possible:
Steam constantly keeps writing to my drive at around 400KiB/s. There are no downloads going on. This never stops. When I try to exit the client, there's always this dialog about stopping a download for Steam Input Configs.

With lsof I've determined that the path being written to are files in .local/share/Steam/steamapps/workshop/downloads/

There's 1919 files in there, each 24-bytes, with a name like state_241100_241100_614902891.patch. These files are either getting written to over and over, or created and deleted constantly. It's just scrolling by too fast for me to see what exactly is happening, except that their number or size doesn't seem to be growing. Steam is obviously trying to download https://steamdb.info/app/241100/, but is failing for some reason.

I have given rwx permissions to every file and/or directory that belongs to steam for all users. I've also tried clearing the download cache, as well as checking the filesystem for errors.

I'm running Steam from the Manjaro repos. I tried both with native and steam runtime libs - doesn't seem to make a difference. Stable vs beta also doesn't make a difference. Also the flatpak version does the same thing.

There's this similar issue on the tracker: https://github.com/ValveSoftware/steam-for-linux/issues/7795 . This person supposedly managed to solve it by moving the location outside ecrypfts, but I'm not currently running any encryption whatsoever, so I feel that's a red herring.

Relevant logs and system information here: https://gist.github.com/MadMinstrel/cdf678f15173667fedef3f46f49c984a
(tell me if something is missing)

#### Steps for reproducing this issue:

1. Install Steam
2. Play a game
3. Wait a while
4. Open a system monitor and note Steam writing to drive

**Comment by seboss666** [comment]
Hello,
I'm noticing the same exact behaviour, I'm also on Manjaro but with Cinnamon desktop.
Strangely, a friend reported some days ago a high cpu usage but on windows. could it be related to some problem that may not be platform specific ?

**Comment by turmoni** [comment]
Adding my vote here - this seems to have been happening pretty reliably for the past couple of weeks, when there's no active download occurring there's a burst of disk activity roughly three times a second, with similar logs. My /home is a ZFS mirror on two disks, one of which is pretty noisy, so I can tell when this is happening (and it's annoying enough to make me close Steam).

Looking into it a bit more, for me this may have been caused by something to do with installing F1 2018 - the timing feels like it fits, and I think it's trying to download https://steamcommunity.com/workshop/filedetails/?id=1382475933 . I have a Logitech G29 wheel which is usually connected, and that doesn't prompt this behaviour, but both an Xbox 360 controller and a Steam Controller do. Unfortunately, uninstalling F1 2018 and restarting Steam doesn't seem to have stopped it from trying to download this.

My `workshop_log.txt` contains many instances of:

```
[2022-02-09 18:20:56] [AppID 241100] Starting Workshop download job (downloading  )
[2022-02-09 18:20:57] [AppID 241100] Get details for item 1382475933 failed : I/O Operation Failed
[2022-02-09 18:20:57] [AppID 241100] Finished Workshop download job : No Error
[2022-02-09 18:20:57] [AppID 241100] Starting Workshop download job (downloading  )
[2022-02-09 18:20:57] [AppID 241100] Get details for item 1382475933 failed : I/O Operation Failed
[2022-02-09 18:20:57] [AppID 241100] Finished Workshop download job : No Error
```

I'm running plain Arch Linux, using the 5.16 -zen kernel

**Comment by TiagoTiago** [comment]
Encrypted home directory by any chance?

**Comment by turmoni** [comment]
Yes, but I'm unsure that's in a way that could have any impact here - I'm using dm-crypt directly on the hard disks, with ZFS using them once they have been unlocked, so unless there's some very weird interaction I don't think anything should know it's encrypted.

**Comment by TiagoTiago** [comment]
I read somewhere there's an issue crossing file systems between /tmp and an encrypted home directory with the way Steam does some stuff; by undoing the home folder encryption, things started working better for me in this aspect. I don't think there's the same issue if you're using full disk encryption though; looks like that specific glitch was when the home folder itself is encrypted (with eCryptfs like how it's offered in Ubuntu and Mint; maybe other approaches as well, dunno).

**Comment by turmoni** [comment]
My /home _is_ a separate filesystem to /tmp, so something like that could still be the cause

**Comment by tinspin** [comment]
Steam is wearing the SSD down writing this file over and over again: steam\steamapps\workshop\appworkshop_241100.acf

Can confirm it stops if I unplug my NES -> USB controller.
